### ISSUE TITLE
Python 有哪些一千行左右的经典练手项目

### ISSUE THREAD

**Issue by mingyun** [mingyun/mingyun.github.io#70]
今天   一如既往最炫酷的云爬虫——造数    给大家带来一批宝贝，大家可以在深夜里独自一人默默研究把玩，也可以在广场上一边遛狗一边和盆友们开心品鉴。
https://www.zhihu.com/question/31830544

经典入门及第三方库

vinta/awesome-python 精心设计的Python学习框架，书籍和软件
nvbn/thefuck 华丽的应用程序更正您以前的控制台命令
pallets/flask、django/django Python的web框架
requests/requests 强大的库，相信你一定学过
ipython/ipython 生产性交互式计算系统
python/cpython Python编程语言官方文档

好玩的项目

warner/magic-wormhole 把文件从一台电脑安全地复制到另一台
pyvideo/pyvideo 和Python相关的视频
7sDream/zhihu-oauth 知乎官方未开放的 OAuth2 接口
errbotio/errbot 最简单和最流行的聊天机器人
fogleman/Minecraft 用Python写的我的世界
mopidy/mopidy 一个可扩展的音乐服务器
Eloston/ungoogled-chromium 可以修改Google Chromium组件
livid/v2ex 在Google App Engine上运行的社区
overviewer/Minecraft-Overviewer 展示高分辨率地图
charlierguo/gmail Google Mail的Pythonic界面
egirault/googleplay-api Google Play非官方的Python API

（有没有想过 用Python 自己写一个Minecraft，筒子们？

网络爬虫

LiuXingMing/SinaSpider 新浪微博爬虫（Scrapy、Redis）
binux/pyspider Python中强大的网络爬虫系统
bowenpay/wechat-spider 微信公众号爬虫 
jhao104/proxy_pool Python爬虫代理IP池(proxy pool)
smicallef/spiderfoot SpiderFoot，开源脚印和情报收集工具。
lining0806/PythonSpiderNotes Python入门网络爬虫之精华版
Germey/Zhihu 崔庆才博主写的知乎爬虫
gnemoug/distribute_crawler 分布式网络爬虫
https://github.com/Chyroc/WechatSogou基于搜狗微信搜索的微信公众号爬虫接口
ResolveWang/weibospider 分布式微博爬虫(PC端抓取)
airingursb/bilibili-user  Bilibili用户爬虫
yanzhou/CnkiSpider 中国知网爬虫

数据相关

donnemartin/data-science-ipython-notebooks 综合型的数据科学教程
sqlmapproject/sqlmap 自动SQL注入和数据库接管工具
mitmproxy/mitmproxy 针对渗透测试员和软件开发人员的交互式TLS功能拦截HTTP代理
rushter/data-science-blogs 很多数据科学的博客
ujjwalkarn/DataSciencePython 通用数据分析和机器学习任务
justmarkham/DAT3 华盛顿的大数据课程
billryan/algorithm-exercise leetcode/lintcode题解
bitly/data_hacks 使用命令行进行数据分析

机器学习

MorvanZhou/tutorials 机器学习相关教程
ahangchen/GDLnotes 谷歌深度学习笔记
tensorflow/models 使用TensorFlow构建模型
astorfi/TensorFlow-World TensorFlow的简单和现成的教程
fchollet/keras Python深度学习库，运行在TensorFlow，Theano或CNTK
josephmisiti/awesome-machine-learning 精心设计的机器学习框架，书籍和软件
songrotek/Deep-Learning-Papers-Reading-Roadmap 深度学习论文阅读路线图
MLWave/Kaggle-Ensemble-Guide  Kaggle组合指南的代码
eriklindernoren/ML-From-Scratch 机器学习模型和算法
humphd/have-fun-with-machine-learning 机器学习与神经网络图像分类的初学者指南
arielf/weight-loss 机器学习符合酮症：如何有效减肥
luispedro/BuildingMachineLearning 本书用Python构建机器学习系统的源代码
